Lib/inspect.py

```diff
@@ -1621,21 +1621,23 @@ def getclosurevars(func):
     global_vars = {}
     builtin_vars = {}
     unbound_names = set()
-    for name in code.co_names:
-        if name in ("None", "True", "False"):
-            # Because these used to be builtins instead of keywords, they
-            # may still show up as name references. We ignore them.
-            continue
-        try:
-            global_vars[name] = global_ns[name]
-        except KeyError:
+
+    for instrn in dis.get_instructions(func):
```

Comment:
I feel a bit uneasy about relying on `dis` here because the bytecode format may change. Also, this may make it harder to use `inspect.py` on alternative implementations such as PyPy.

I don't have a better idea for implementing `getclosurevars` correctly though.